pkg-manifest/exportable-manifest/src/index.ts

```diff
@@ -100,3 +108,37 @@ async function makePublishDependency (depName: string, depSpec: string, dir: str
   }
   return depSpec
 }
+
+async function makePublishPeerDependency (depName: string, depSpec: string, dir: string, modulesDir?: string) {
+  if (!depSpec.includes('workspace:')) {
+    return depSpec
+  }
+
+  // Dependencies with bare "*", "^", "~",">=",">","<=",< versions
+  const workspaceSemverRegex = /workspace:([\^~*]|>=|>|<=|<)/
+  const versionAliasSpecParts = workspaceSemverRegex.exec(depSpec)
+
+  if (versionAliasSpecParts != null) {
+    modulesDir = modulesDir ?? path.join(dir, 'node_modules')
+    const manifest = await resolveManifest(depName, modulesDir)
+
+    const [,semverRangGroup] = versionAliasSpecParts
+
+    const semverRangeToken = semverRangGroup !== '*' ? semverRangGroup : ''
+
+    return depSpec.replace(workspaceSemverRegex, `${semverRangeToken}${manifest.version}`)
+  }
+
+  if (depSpec.includes('workspace:./') || depSpec.includes('workspace:../')) {
+    const manifest = await resolveManifest(depName, path.join(dir, depSpec.replace('workspace:', '')))
+
+    if (manifest.name !== depName) {
+      throw new PnpmError('DO_NOT_SUPPORT_ALIAS_IN_PEER_DEPENDENCIES', `Workspace protocol for peerDependencies does not support aliasing of "${depName}" to "${manifest.name}"`)
```

Comment:
`DO_NOT_SUPPORT_ALIAS_IN_PEER_DEPENDENCIES` is there an expectation of format for the errors?

Reply:
[Took this out. Decided no need ](https://github.com/pnpm/pnpm/pull/7633#discussion_r1508261449)